### ISSUE TITLE
[Issue] CVE-2023-23397 script - Failed to clear property

### ISSUE THREAD

**Issue by GeoCookie** [microsoft/CSS-Exchange#1610]
**Describe the issue**
CVE-2023-23397 script version 23.03.17.2033

**Expected behavior**
When running the script in cleanup mode with "ClearProperty" parameter, the cleanup failed
With the parameter "ClearItem", the object is correctly deleted.

**Script Output**
![image](https://user-images.githubusercontent.com/43962716/226557275-4ec0ab76-8d8d-413d-ba47-604951db71e2.png)
Unfortunately, the script doesn't return any explanation

**Additional context**
Exchange servers are installed with french language (system and UI). Same for Exchange 2019.

I have followed the prerequisites (throttling policy and impersonation role)

My admin account has a mailbox and is member of these accounts
![image](https://user-images.githubusercontent.com/43962716/226558211-e355a82a-545a-4063-b859-e6ab60ffae4d.png)
It also has the correct throttling policy
![image](https://user-images.githubusercontent.com/43962716/226558316-a679b96b-6d57-4b84-97f4-fdfee42ced09.png)

The csv file I am using for the tests
![image](https://user-images.githubusercontent.com/43962716/226558437-ec7288b5-b29e-48ac-aa10-6cfffbafc53b.png)

The cleanup command I am running directly from an Exchange server
`.\CVE-2023-23397.ps1 -Environment Onprem -DLLPath .\Microsoft.Exchange.WebServices.dll -SkipVersionCheck -CleanupAction ClearProperty -CleanupInfoFilePath .\AuditResults_20230321_090848.csv -EWSServerURL https://127.0.0.1/EWS/Exchange.asmx -IgnoreCertificateMismatch`
I have tried without the EWSUrl parameter, same result.

Please note that with a previous version of the script, I had these error messages (a quite bit explicit)
![image](https://user-images.githubusercontent.com/43962716/226559471-8a3b0851-4f15-4fb1-8806-2e3fc43cab54.png)
Translated error message : _Exception calling 'Update' with '1' argument(s): 'At least one recipient is invalid., A message cannot be sent because it does not contain a recipient. »_

**Comment by GeoCookie** [comment]
I hope I haven't missed something obvious. Feel free to ask me anything.
Thank you for the script and for your help

**Comment by dpaulson45** [comment]
@GeoCookie I am not sure why you are still facing this issue. It should have been resolved with this PR #1600 

Based off the error message, it seems like we are not determining that the entry is a `Calendar` type. But the image does report it as one.

**Comment by bill-long** [comment]
It looks like this occurs when the CSV is opened in Excel on a machine where the default delimiter is not a comma. When choosing Save As, Excel sees the file as having only one column and surrounds the entire line with a quote.

We can undo this with some PowerShell script. To fix an existing file in this state:

```powershell
$inputFile = "C:\bad.csv"
$newFileName = "C:\fixed.csv"
$importedData = Import-Csv $inputFile
$headerLine = ($importedData[0] | Get-Member | ? { $_.MemberType -eq "NoteProperty" }).Name
$headerLine | Out-File $newFileName
$importedData | % { $_."$headerLine" } | Out-File $newFileName -Append
```

We might be able to fix these on the fly in the CVE script by doing this:

```powershell
$inputFile = "C:\bad.csv"
$importedData = Import-Csv $inputFile
$headerLine = ($importedData[0] | Get-Member | ? { $_.MemberType -eq "NoteProperty" }).Name
$fixedData = $importedData | % { $_."$headerLine" } | ConvertFrom-Csv -Header ($headerLine.Split(","))
```

**Comment by GeoCookie** [comment]
I tried again with two new calendar appointments and two new tasks.
I ran the script one time in audit mode, change the Cleanup values to "Y" with notepad and ran the script in "CleanProperty" mode. This time, everything went fine. All four objects are correctly cleaned!

If I run the script in cleanup mode a second time with the same csv file, I got the same errors.
![image](https://user-images.githubusercontent.com/43962716/226628078-1d35efe4-d771-42a1-9da2-a0e28f013125.png)

So maybe at some point, the script correctly removed the property and I didn't check correctly the appointments on Outlook. But I am sure the script throws an error at each run.

Anyway, I am now able to run the script against our mailboxes.

Thanks a lot for your help!
